src/rgw/rgw_rest_pubsub.cc

```diff
@@ -413,23 +413,33 @@ void RGWPSCreateTopicOp::execute(optional_yield y) {
     }
   }
 
-  // don't add a persistent queue if we already have one
+  // if we already have an existing persistent queue - do nothing - resharding not supported
+  // if we don't have a persistent queue already, just make shards or a single based on dest's num_shards from config
+
   const bool already_persistent = topic && topic_needs_queue(topic->dest);
+  std::vector<std::string> q_names; 
   if (!already_persistent && topic_needs_queue(dest)) {
     // initialize the persistent queue's location, using ':' as the namespace
     // delimiter because its inclusion in a TopicName would break ARNs
+    // new persistent topics to have shards - loading num_shards from config for sharding persistent bucket notifications
+    dest.num_shards = s->get_cct()->_conf.get_val<uint64_t>("rgw_bucket_persistent_notif_num_shards");    
     dest.persistent_queue = string_cat_reserve(
         get_account_or_tenant(s->owner.id), ":", topic_name);
-
-    op_ret = driver->add_persistent_topic(this, y, dest.persistent_queue);
-    if (op_ret < 0) {
-      ldpp_dout(this, 1) << "CreateTopic Action failed to create queue for "
-                            "persistent topics. error:"
-                         << op_ret << dendl;
-      return;
+    for(const auto& q: dest.get_queue_names()){
+      op_ret = driver->add_persistent_topic(this, y, q);
+      if (op_ret < 0) {
+        ldpp_dout(this, 1) << "CreateTopic Action failed to create queue/shards for "
+                              "persistent topics. error:"
+                          << op_ret << dendl;
+        return;
+      }
     }
+    ldpp_dout(this, 20) << "Successfully created " <<dest.num_shards<<" shards for topic: " + topic_name<<dendl;
   } else if (already_persistent) {  // redundant call to CreateTopic
+    //no resharding of existing topics for persistent bucket notifications
     dest.persistent_queue = topic->dest.persistent_queue;
+    dest.num_shards = topic->dest.num_shards;
+    ldpp_dout(this, 20) << "Topic already has persistent shards, skipping creation. num_shards: " <<topic->dest.num_shards<<dendl;
```

Comment:
i think that if the conf num of shards is different, we can emit a different log message ("warning")